pulp_cli/__init__.py

```diff
@@ -1 +1,19 @@
-from pulpcore.cli.common import main  # noqa: F401
+from types import ModuleType
+from typing import Any, Callable, Dict
+
+import pkg_resources
+
+
+def main() -> Any:
+    ##############################################################################
+    # Load plugins
+    # https://packaging.python.org/guides/creating-and-discovering-plugins/#using-package-metadata
+    discovered_plugins: Dict[str, ModuleType] = {
+        entry_point.name: entry_point.load()
+        for entry_point in pkg_resources.iter_entry_points("pulp_cli.plugins")
+    }
+    _main: Callable[[], Any] = discovered_plugins["common"].main
```

Comment:
What happens if a plugin also declares a common folder? Does it overwrite it?

Reply:
No problem unless the plugin is named "common". Maybe i am missing something.
Are you saying that multiple packages could define the same `entry_point` in `"pulp_cli.plugins"`. I suspect this would break the system already today.

Reply:
Also in the long run we might pull the main function out of the common plugin, or make common not-a-plugin. wdyt?

Reply:
Right now all the other *external* plugins are separate packages containing just that plugin, but I wonder what would happen if you had a package that contained multiple plugins with one of them also being named "common". It would probably break today as well, but I think this question could be expanded to any name conflicts between the packages. How badly would it break and is this a scenario we should be concern about?

I think having common not be a plugin makes sense. All the other plugins rely on it, so it isn't optional like the "plugin" nomenclature would suggest.

Reply:
But it was not so straight forward to do, so i started small.